src/components/networked-hand-controls.js

```diff
@@ -336,29 +336,34 @@ AFRAME.registerComponent('networked-hand-controls', {
     // handling for older controller components (like vive-controls) that lack the cleaner hooks 
     // to block local post tracking. We allow pose tracking to start (so we can get loaded model), 
     // and then stop it. Note: we do not attempt to handle older tracked-controls-webvr.
-    let counter = 0;
-    let disableInterval = setInterval(() => {
-      counter++;
-      if (this.el.components['tracked-controls-webxr']) {
-        // there is no remove on this function in current iteration, so we kill it with pause and tick first.
-        this.el.components['tracked-controls-webxr'].pause(); 
-        this.el.components['tracked-controls-webxr'].tick = () => {};
-        this.el.removeAttribute('tracked-controls-webxr'); // just some cleanup for inspector, etc.
+    let intervalCounter = 0;
 
-        clearInterval(disableInterval);
-        this.el.play();
-      }
-      if (counter > 1000) {
-        // 25 seconds, don't want the possiblity of this running forever in the background
-        clearInterval(disableInterval);
+    const disablePostTracking = () => {
+      intervalCounter++;
+      if (intervalCounter > 1000) { // 7-17 seconds, assuming 140hz-60hz refresh rate
+        // don't want the possiblity of this running forever in the background
         try {
+          this.el.components[this.data.controllerComponent].pause();
           this.el.components[this.data.controllerComponent].remove();
         } catch(e) {
           console.error("error removing controller component", e);
         }
         throw new Error("never found tracked-controls-webxr, cancelling interval that was waiting to remove it and removing controller components")
       }
-    },25)
+      else if (this.el.components['tracked-controls-webxr']) {
+        // there is no remove on this component in current version of A-Frame, so we kill it with pause and tick first.
+        this.el.components['tracked-controls-webxr'].pause();  // remove its event listeners
+        this.el.components['tracked-controls-webxr'].tick = () => {}; // remove on-tick pose tracking in the component
+        this.el.components['tracked-controls-webxr'].remove(); // in case support is added in the future
+        this.el.removeAttribute('tracked-controls-webxr'); // just some cleanup for inspector, etc.
+        this.el.play();
+      }
+      else {
+        requestAnimationFrame(disablePostTracking);
+      }
+    };
+
+    requestAnimationFrame(disablePostTracking);
```

Comment:
You probably meant Pose instead of Post? So disablePoseTracking?

You didn't have an issue with window.requestAnimationFrame? I remember there is an issue (rendering a pdf page on a canvas) with window.requestAnimationFrame not being executed in an XR session.
If it's working properly that's great, otherwise you can consider setting a flag on the component and execute disablePoseTracking if the flag is set in the tick method. You can find this pattern in the raycaster component with the dirty flag for example, there the refreshObjects is called in checkIntersections not in tick but it's the almost the same.

Reply:
> You probably meant Pose instead of Post? So disablePoseTracking?

yup!

> You didn't have an issue with window.requestAnimationFrame?

I can only test this by using an emulator. It seemed to work, but digging into it more, I see that it _may_ cause issues, rarely. I've updated the code to handle either case, and was able to document both window.requestAnimationFrame as well as xrSession.requestAnimationFrame being used with new code.

Reply:
Ok, looking at the whole code again, I think this is too hacky to check if a component has been set by regularly checking for it like that to then remove it. I think we better should check for the componentinitialized event and then call removeAttribute if this is the component name we're waiting for. You have an example in the animation component here:
https://github.com/aframevr/aframe/blob/837023cca1a04b408b4acde4c09823e09dc58e6d/src/components/animation.js#L452-L459

Reply:
Sure, I understand the feeling. I hadn't noticed that event before, that does look nice.

I just wrote up a version that uses it, unfortunately I also have to use a hack with it; if I don't do a 1 millisecond setTimeout, A-Frame seems to have a weird bug that locks it in an infinite loop if I remove in the event listener itself.